### PR TITLE
T/702 Selection breaks ViewAttributeElement created by marker conversion

### LINKED ISSUE #702: Markers: Selection breaks markers ViewAttributeElement.
The issue is simple. Consider some letters in a paragraph and a marker on them. They could be converter to something like this in view:
`<p>ab<span class="marker">cde</span>fg</p>`.
Now, selection is placed after `d`:
`<p>ab<span class="marker">cd</span><fillingChar>|<span class="marker">e</span>fg</p>`.
This happens because selection does not have marker's attributes.

Also weird thing happens next to space:
```
<p>Lorem ipsu<span class="h-yellow">m</span> <fillingChar>|<span class="h-yellow"> dolor s</span>it amet.</p>
```
This makes the space disappear. Somehow, this bug does not happen in following scenario
```
<p><strong>abc def ghi</strong></p>
```
When I place cursor after `c` or before `d` and "unbold" the space is correctly moved to left or right `em`.

I see two solutions:
1. Mark (pun not indented) view elements created by markers so they won't be broken by selection converter.
2. Use `ViewContainerElement` instead of `ViewAttributeElement` but this may have unexpected and incorrect impact on `view.Writer` and conversion. I'd advise against this.
3. Add attributes on selection (somehow).

### PR DESCRIPTION
**Pull request by scofalik**
Fixes #702 .
Bases on PR #774 .

Fixes also:
#703

### CODE REVIEW

**Comment by pjasiun** [comment]
I don't like this set on the selection with all markers selection is in. It works only on collapsed selection and needs to be always synchronized. Also, as @scofalik note: it does not fit well for Non-Live (Dead?) Selection. In fact, since it's only used by converters which do not listen on any changes there but get markers on demand, it could be replaced with the method in `MarkersCollection` which returns all markers a position is in (such method might be useful in other cases). Then converter may use this method.

File: src/conversion/model-selection-to-view-converters.js [hunk 1]
```diff
@@ -145,27 +145,54 @@ export function convertCollapsedSelection() {
...
-		conversionApi.viewSelection.removeAllRanges();
+		convertSelectionPosition( data.selection, conversionApi.viewSelection, viewElement, consumable, 'selectionMarker:' + data.name );
+	};
+}
 
-		const viewElement = elementCreator instanceof ViewElement ?
-				elementCreator.clone( true ) :
-				elementCreator( data.value, data, selection, consumable, conversionApi );
+// Helper function for `convertSelectionAttribute` and `convertSelectionMarker`, which perform similar task.
+function convertSelectionPosition( modelSelection, viewSelection, viewElement, consumable, consumableName ) {
```
Comment (pjasiun) [inline src/conversion/model-selection-to-view-converters.js]: Does not method convert "full" selection (with attribute and markers)? If not then the name is misleading. Maybe `wrapSelectionPosition`?

File: src/view/writer.js [hunk 2]
```diff
@@ -887,6 +887,8 @@ function wrapChildren( parent, startOffset, endOffset, attribute ) {
 	let offsetChange = 0;
 
 	for ( let position of wrapPositions ) {
+		position.offset -= offsetChange;
```
Comment (pjasiun) [inline src/view/writer.js]: Qu'est-ce?

Comment (scofalik) [inline src/view/writer.js]: It is explained by commit message.

This got removed, probably due to mistake and was causing a bug. This exact line of code was there long time ago in repo history.

Comment (pjasiun) [inline src/view/writer.js]: Please add comment then.

Comment (scofalik) [inline src/view/writer.js]: But I don't know why this is like this. I just recovered a line of code that was mistakenly removed and added a test. What am I going to comment? `// Please do not remove it by mistake again.` ?

**Pull request merged**